app/src/main/java/com/eveningoutpost/dexdrip/watch/miband/MiBandService.java

```diff
@@ -164,20 +151,120 @@ public QueueMessage(String functionName, String message_type, String message, St
+
+
+        l.add(new StatusItem("Model", MiBand.getModel()));
+        l.add(new StatusItem("Software version", MiBand.getVersion()));
+
+        l.add(new StatusItem("Mac address", MiBand.getMac()));
+        l.add(new StatusItem("Connected", II.isConnected ? gs(R.string.yes) : gs(R.string.no)));
+        l.add(new StatusItem("Is authenticated", MiBand.isAuthenticated() ? gs(R.string.yes) : gs(R.string.no)));
+        if (II.isConnected) {
+            int levelInPercent = batteryInfo.getLevelInPercent();
+            String levelInPercentText;
+            if (levelInPercent == 1000)
+                levelInPercentText = "Unknown";
+            else
+                levelInPercentText = levelInPercent + "%";
+            l.add(new StatusItem("Battery", levelInPercentText));
+        }
+        if (II.wakeup_time != 0) {
+            final long till = msTill(II.wakeup_time);
+            if (till > 0) l.add(new StatusItem("Wake Up", niceTimeScalar(till)));
+        }
+
+        if (bgWakeupTime != 0) {
+            final long till = msTill(bgWakeupTime);
+            if (till > 0) l.add(new StatusItem("Next time update", niceTimeScalar(till)));
+        }
+
+        l.add(new StatusItem("State", II.state));
+
+        final int qsize = II.getQueueSize();
+        if (qsize > 0) {
+            l.add(new StatusItem("Queue", qsize + " items"));
+        }
+
+        return l;
+    }
+
     private Class getPrefBinder() {
-        MiBand.MiBandType type = MiBand.getMibandType();
-        if (type == MI_BAND2)
+        MiBandType type = MiBand.getMibandType();
+        if (type == MiBandType.MI_BAND2)
             return Miband2PrefBinding.class;
         else return Miband3_4PrefBinding.class;
     }
 
     @Override
     public void onCreate() {
+
+        statusReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                final String iob = intent.getStringExtra("iob");
+                if (iob != null) {
+                    statusIOB = iob;
+                }
+            }
+        };
+        LocalBroadcastManager.getInstance(this).registerReceiver(statusReceiver,
+                new IntentFilter(Intents.HOME_STATUS_ACTION));
         super.onCreate();
     }
 
+    @Override
+    public void onDestroy() {
+        try {
+            LocalBroadcastManager.getInstance(this).unregisterReceiver(statusReceiver);
+        } catch (Exception e) {
+            UserError.Log.e(TAG, "Exception unregistering broadcast receiver: " + e);
+        }
+    }
+
     private boolean readyToProcessCommand() {
         boolean result = I.state.equals(MiBandState.SLEEP) || I.state.equals(MiBandState.CLOSED) || I.state.equals(MiBandState.CLOSE) || I.state.equals(MiBandState.INIT) || I.state.equals(MiBandState.CONNECT_NOW);
+        if (!result && I.state.equals(MiBandState.AUTHORIZE_FAILED) && MiBandType.supportPairingKey(MiBand.getMibandType())) {
```

Comment:
I cannot follow the logic here. Why is it possible to process a command if the state is `AUTHORIZE_FAILED`?